common/src/chain/config/mod.rs

```diff
@@ -467,8 +468,13 @@ impl ChainConfig {
 
     /// The maximum offset of time from the current time the timestamp of a new block can be
     #[must_use]
-    pub fn max_future_block_time_offset(&self) -> &Duration {
-        &self.max_future_block_time_offset
+    pub fn max_future_block_time_offset(&self, height: BlockHeight) -> Duration {
+        self.max_future_block_time_offset.unwrap_or_else(|| {
+            match self.as_ref().chainstate_upgrades().version_at_height(height).1.htlc_activated() {
+                HtlcActivated::Yes => MAX_FUTURE_BLOCK_TIME_OFFSET,
+                HtlcActivated::No => DEFAULT_MAX_FUTURE_BLOCK_TIME_OFFSET,
+            }
+        })
```

Comment:
I makes sense to leave a comment that we've switched to the new offset at the same height where we've enabled htlc, and that otherwise they have nothing to do with each other (because judging from this code it seems like they do).